### PR TITLE
[CFM-1150] adjust page template for Project pages (header & footer)

### PR DESCRIPTION
**Pull request by anvmn**
https://issuetracker.amplexor.com/jira/browse/CFM-1150

### CODE REVIEW

**Comment by anvmn** [comment]
selection_079

**Comment by anvmn** [comment]
selection_085

**Comment by anvmn** [comment]
@ordavidil 
Travis failed on 'Group Dashboard'. Not related to project page.

**Comment by ordavidil** [comment]
@anvmn 
re-roll with develop

**Comment by anvmn** [comment]
@ordavidil 
Re-rolled.

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/c4m_features_footer.module [hunk 1]
```diff
@@ -63,6 +63,15 @@ function c4m_features_footer_theme() {
     'file'  => 'c4m_features_footer.theme.inc',
     'path'  => drupal_get_path('module', 'c4m_features_footer') . '/includes',
   );
+  $theme['c4m_footer_bar_project'] = array(
+    'variables' => array(
+      '$front_page' => '',
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/c4m_features_footer.module]: => `'front_page'`

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/c4m_features_footer.module [hunk 2]
```diff
@@ -63,6 +63,15 @@ function c4m_features_footer_theme() {
...
+      '$project_page' => '',
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/c4m_features_footer.module]: => `'project_page'`

File: project/profiles/capacity4more/themes/c4m/kapablo/theme/preprocess.inc [hunk 3]
```diff
@@ -98,6 +98,21 @@ function kapablo_preprocess_block(&$variables) {
  * @see page.tpl.php
  */
 function kapablo_preprocess_page(&$variables) {
+  // Add information about OG context type.
+  $context = og_context();
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/theme/preprocess.inc]: @anvmn please use:
```php
  $variables['page']['is_og_project'] = FALSE;
  if (($group_type = c4m_og_get_group_type(og_context())) && $group_type == 'project') {
    // Determine if the current context is a project.
    $variables['page']['is_og_project'] = TRUE;
  }
```

File: project/profiles/capacity4more/themes/c4m/kapablo/templates/page.tpl.php [hunk 4]
```diff
@@ -93,103 +93,109 @@
     </section>
   <?php endif; ?>
 
-  <!--  TOP MENU -->
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/templates/page.tpl.php]: @anvmn 
Can't we just use a different template instead of changing `page.tpl.php` ?

Comment (anvmn) [inline project/profiles/capacity4more/themes/c4m/kapablo/templates/page.tpl.php]: @ordavidil
We have group and project content types, that require different page templates.
However, they're both OG group (node).
The difference between them is their $og_group->type.
Couldn't find a way to assign different templates to differentiate between the 2.

Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/templates/page.tpl.php]: @nirgn975 
Please give some help over here of how to implements a _hook_suggestion_
thanks.

File: project/profiles/capacity4more/themes/c4m/kapablo/templates/page--og--project.tpl.php [hunk 5]
```diff
@@ -0,0 +1,242 @@
+<?php
+
+/**
+ * @file
+ * Theme suggestion implementation to display a page for OG type = 'project'.
+ *
+ * The doctype, html, head and body tags are not in this template. Instead they
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/templates/page--og--project.tpl.php]: Please remove all the docs besides the first line.

File: project/profiles/capacity4more/themes/c4m/kapablo/theme/preprocess.inc [hunk 6]
```diff
@@ -96,8 +96,15 @@ function kapablo_preprocess_block(&$variables) {
  * Implements theme_preprocess_page().
  *
  * @see page.tpl.php
+ * @see page--og--project.tpl.php
  */
 function kapablo_preprocess_page(&$variables) {
+  // Determine if the current context is a project.
+  if (($group_type = c4m_og_get_group_type(og_context())) && $group_type == 'project') {
+    // Instruct to use a page template designated for project OG type.
+    $variables['theme_hook_suggestion'] = 'page__og__project';
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/theme/preprocess.inc]: try using `$variables['theme_hook_suggestion'][] = 'page__og__project';`

File: project/profiles/capacity4more/themes/c4m/kapablo/templates/page.tpl.php [hunk 7]
```diff
@@ -122,21 +122,20 @@
 
           <div class="header-actions">
 
-            <span class="header-actions--wrapper">
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/templates/page.tpl.php]: please revert the changes in this template.

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php [hunk 8]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+/**
+ * @file
+ * Template for presenting OG project pages footer.
+ */
+?>
+
+<div class="row">
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php]: we shouldn't use `row` without a `col-...` inside.

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  <ul class="c4m-footer-bar clearfix">
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php]: when/if removing the `row` class from the parent element, check if we still need the `clearfix` in here

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    <li class="first">
+      <a class="logo" href="<?php print $front_page; ?>"
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php]: Let's render the link on the preprocess function.

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+         title="<?php print t('Home'); ?>">
+        <img src="<?php print $logo; ?>" alt="<?php print t('Home'); ?>"/>
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php]: same thing, render the image on the preprocess function.

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+      </a>
+    </li>
+    <li class="c4m-footer-bar-project">
+      <div class="text-uppercase">
+        Explore more projects on
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php]: should we put this one inside a `t()` function ?

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+        <a href="<?php print $project_page; ?>" class="text-orange">Capacity4dev ></a>
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php]: render the link on the preprocess function.

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+        <br/>
+        <a href=" https://ec.europa.eu/europeaid/home_en" class="">International cooperation and development ></a>
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php]: 1. Remove the `class` since it's empty.
2. Remove the extra space from the beginning of the `href`.
3. Let's use `&gt;` instead of `>`.

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+      <a class="logo" href="<?php print $front_page; ?>"
+         title="<?php print t('Home'); ?>">
+        <img src="<?php print $logo; ?>" alt="<?php print t('Home'); ?>"/>
+      </a>
+    </li>
+    <li class="c4m-footer-bar-project">
+      <div class="text-uppercase">
+        <?php print t('Explore more projects on') ?>
+        <a href="<?php print $project_page; ?>" class="text-orange">
+          <?php print t('Capacity4dev &gt') ?>
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php]: `&gt` should be out of the `t()` function.

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+        </a>
+        <br/>
+        <a href="https://ec.europa.eu/europeaid/home_en">
+          <?php print t('International cooperation and development &gt') ?>
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php]: `&gt` should be out of the `t()` function.

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php (hunk 13 above)
Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php]: @ordavidil 
When you say render image/link at preprocess function.
You suggest I don't use c4m-features-footer-bar-project.tpl.php anymore?
Move it's html to project page tempate?

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/includes/c4m_features_footer.theme.inc [hunk 17]
```diff
@@ -58,6 +65,23 @@ function theme_c4m_footer_bar(array $variables) {
...
 /**
+ * Theme function OG Project pages footer - uses project footer template.
+ *
+ * @return html
+ *   Rendered html.
+ */
+function theme_c4m_footer_bar_project() {
+  $variables = array(
+    'front_page' => base_path(),
+    'project_page' => base_path() . 'projects',
```
Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/includes/c4m_features_footer.theme.inc]: use `l()` here

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/includes/c4m_features_footer.theme.inc [hunk 18]
```diff
@@ -58,6 +65,38 @@ function theme_c4m_footer_bar(array $variables) {
...
+    'attributes' => array()
+  );
+  $europaid_link = l(t('International cooperation and development') . ' >', 'https://ec.europa.eu/europeaid/home_en');
+
+  $options['attributes']['class'] = array('text-orange');
+  $project_link = l(t('Capacity4dev') . ' >',  base_path() . 'projects', $options);
+
+  $options['attributes']['title'] = 'Home';
+  $options['attributes']['class'] = array('logo');
+  $logo_img = '<img src="' . file_create_url(drupal_get_path('theme', 'kapablo') . '/project_logo.png') . '" alt="' . t('Home') . '"/>';
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/includes/c4m_features_footer.theme.inc]: We probably want to use 
https://api.drupal.org/api/drupal/includes!theme.inc/function/theme_image/7.x

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/includes/c4m_features_footer.theme.inc [hunk 19]
```diff
@@ -58,6 +65,38 @@ function theme_c4m_footer_bar(array $variables) {
...
+    'alias' => TRUE,
+    'html' => TRUE,
+    'attributes' => array()
+  );
+  $europaid_link = l(t('International cooperation and development') . ' >', 'https://ec.europa.eu/europeaid/home_en');
+
+  $options['attributes']['class'] = array('text-orange');
+  $project_link = l(t('Capacity4dev') . ' >',  base_path() . 'projects', $options);
+
+  $options['attributes']['title'] = 'Home';
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/includes/c4m_features_footer.theme.inc]: => `t('Home')`

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module [hunk 20]
```diff
@@ -1253,3 +1253,28 @@ function c4m_og_get_user_administered_groups_count($uid, $node_type) {
...
+ *
+ * @return bool
+ *   Wither current OG group type matches the input.
+ */
+function c4m_og_is_of_group_type($type) {
+  $context = og_context();
+  if (!$context) {
+    return FALSE;
+  }
+  else {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: Since we return early we don't need the `else`

**Comment by jover** [comment]
Following the designs, the footer of the project page should have a lot more margin (top & bottom) than the default footer.
project-programme footer

**Comment by ordavidil** [comment]
@jover 
Is this the only issue here ?

**Comment by jover** [comment]
@ordavidil Not sure if this the only issue here, because the title says "header & footer", but currently I see only commits related to the footer.

**Comment by ordavidil** [comment]
@jover 
check this:
https://github.com/capacity4dev/capacity4more/pull/1005/files#diff-94a04c280a13479cdbb262b438965a6cR9

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php [hunk 21]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+/**
+ * @file
+ * Template for presenting OG project pages footer.
+ */
+?>
+
+<div >
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_footer/theme/c4m-features-footer-bar-project.tpl.php]: @anvmn 
remove the extra space inside the `<div>`

File: project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_footer.scss [hunk 22]
```diff
@@ -36,6 +36,23 @@
...
+      font-size: 11px;
+    }
+
+    a.logo-project {
+      img {
+        height: 27px;
+      }
+    }
+
+    div> a.text-orange {
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_footer.scss]: @anvmn please use _SCSS_ convention

File: project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_footer.scss [hunk 23]
```diff
@@ -36,6 +36,23 @@
...
+
+    &.c4m-footer-bar-project {
+      text-align: left;
+      border: none;
+      vertical-align: middle;
+      font-size: 11px;
+    }
+
+    a.logo-project {
+      img {
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_footer.scss]: Add line spacing above

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module [hunk 24]
```diff
@@ -1253,3 +1253,26 @@ function c4m_og_get_user_administered_groups_count($uid, $node_type) {
...
 }
+
+/**
+ * Compares the OG type (project|group) to input.
+ *
+ * @param string $type
+ *   The group to get the type from.
+ *
+ * @return bool
+ *   Wither current OG group type matches the input.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: => `Whether`

**Comment by anvmn** [comment]
@jover , @ordavidil 
Check out the difference between group and project header:

selection_017
selection_019

File: project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_footer.scss [hunk 25]
```diff
@@ -44,14 +44,24 @@
...
+          height: 27px;
+        }
       }
     }
 
-    div> a.text-orange {
-      color: $color-highlight;
+    div {
+
+      a {
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_footer.scss]: @anvmn 
isn't the `>` needed in here ?

**Comment by anvmn** [comment]
@ordavidil 
Don't think so:

selection_021

File: project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_footer.scss (hunk 25 above)
Comment (anvmn) [inline project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_footer.scss]: @ordavidil 
Don't think so:

selection_021

**Comment by anvmn** [comment]
Project foot margin increased:

selection_022

**Comment by anvmn** [comment]
Tablet:

selection_023

Mobile:

selection_024

**Comment by ordavidil** [comment]
@anvmn 
It seems that on mobile we loose the background, is it by design ?

**Comment by anvmn** [comment]
@ordavidil 
We don't loose it:

selection_025

**Comment by ordavidil** [comment]
@anvmn 
I'm talking about the (blueish) background of the footer.

Tablet / Desktop:
selection_250

Mobile:
selection_249

**Comment by anvmn** [comment]
@ordavidil 
Yes. Per Design:

selection_026
